site/tutorials/keptn-multistage-qualitygates-09.md

```diff
@@ -128,31 +128,38 @@ In the `shipyard.yaml` shown above, we define two stages called *hardening* and
 ## Create a service
 Duration: 2:00
 
-After creating the project, we can continue by onboarding the *helloserver* as a service to your project using the `keptn create service` and `keptn add-resource` commands. You need to pass the project where you want to create the service, as well as the Helm chart of the service.
-For this purpose we need the helm charts as a tar.gz archive. To archive it use following command:
+After creating the project, we can continue by onboarding the *helloservice* as a service to your project using the `keptn create service` and `keptn add-resource` commands. You need to pass the project where you want to create the service, as well as the Helm chart of the service.
+For this purpose we need the helm charts as a tar.gz archive. Our repository already contains a tar.gz archive of our helm chart but normally you could use the following command for the conversion:
 
 <!-- command -->
 ```
-tar cfvz ./helm-charts/helloserver.tgz ./helm-charts/helloserver
+tar cfvz ./helm/helloservice.tgz ./helm/helloservice
 ```
 
 Then the service can be created:
 <!-- command -->
 ```
-keptn create service helloserver --project="pod-tato-head"
-keptn add-resource --project="pod-tato-head" --service=helloserver --all-stages --resource=./helm-charts/helloserver.tgz --resourceUri=helm/helloserver.tgz
+keptn create service helloservice --project="podtatohead"
+keptn add-resource --project=podtatohead --service=helloservice --all-stages --resource=helm/helloservice.tgz --resourceUri=helm/helloservice.tgz
 ```
 
 After onboarding the service, tests (i.e., functional- and performance tests) need to be added as basis for quality gates. We are using JMeter tests, as the JMeter service comes "batteries included" with our Keptn installation.
 
 <!-- command -->
 ```
-keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/load.jmx --resourceUri=jmeter/load.jmx
-keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/jmeter.conf.yaml --resourceUri=jmeter/jmeter.conf.yaml
+keptn add-resource --project=podtatohead --stage=hardening --service=helloservice --resource=jmeter/load.jmx --resourceUri=jmeter/load.jmx
+keptn add-resource --project=podtatohead --stage=hardening --service=helloservice --resource=jmeter/jmeter.conf.yaml --resourceUri=jmeter/jmeter.conf.yaml
 ```
 
 Now each time Keptn triggers the test execution, the JMeter service will pick up both files and execute the tests.
 
+Lastly before deploying our first build we also need to define the endpoint the requests should be made against:
+
+<!-- command -->
+```
+keptn add-resource --project=podtatohead --service=helloservice --stage=hardening --resource=helm/endpoints.yaml --resourceUri=helm/endpoints.yaml
+```
+
```

Comment:
Why is this needed? It's only used for stage hardening and only contains
```
deploymentURIsLocal:
  - "http://helloservice.podtatohead-hardening:80" 
```
Can you please verify that the tutorial works without it?

Reply:
Just got information that this is required for user-managed helm charts.

However, I believe we should consider adding an endpoints.yaml file for all stages - see https://github.com/keptn/examples/issues/201